src/incomingDelivery/controller.js

```diff
@@ -517,6 +519,48 @@ function getAllReceived(req, res) {
     "getting all received incoming deliveries"
   );
 }
+
+function setCanceled(req, res) {
+  ExpressHandler(
+    async () => {
+      const { _id, reason } = req.body;
+
+      if (!reason) return HTTPError(`Reason is required.`, 400);
+
+      const incomingDelivery = await IncomingDelivery.findById(_id);
+
+      if (!incomingDelivery)
+        return HTTPError(`Incoming Delivery doesn't exist.`);
+
+      if (incomingDelivery.canceled)
+        return HTTPError(`Incoming Delivery already canceled.`, 400);
+
+      if (!incomingDelivery.receivedBy)
```

Comment:
If this does not exist, that means it has not been received yet. So why cant it be canceled? If it does exist, it has already been received and can't be canceled. Maybe thats the check we need?

Reply:
Sure.